ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -146,14 +152,31 @@ impl<'a> Assemble<'a> {
         #[cfg(debug_assertions)]
         m.assert_well_formed();
 
-        let ra = Box::new(SpillAllocator::new(StackDirection::GrowsDown));
+        let mut inst_vals_alive_until = vec![InstIdx::new(0).unwrap(); m.insts_len()];
+        for iidx in m.iter_all_inst_idxs() {
+            let inst = m.inst_all(iidx);
+            inst.map_packed_operand_locals(m, &mut |x| {
+                inst_vals_alive_until[usize::from(x)] = iidx;
+            });
+        }
+
+        // FIXME: this is a hack.
```

Comment:
What is this hack for?

Reply:
This is basically a bad version of "labels with parameters".